internal/store/atlas/store.go

```diff
@@ -0,0 +1,498 @@
+// Copyright 2020 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package atlas
+
+//go:generate mockgen -destination=../../mocks/atlas/store.go -package=atlas github.com/mongodb/mongodb-atlas-cli/internal/store/atlas CredentialsGetter
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/mongodb-forks/digest"
+	"github.com/mongodb/mongodb-atlas-cli/internal/config"
+	"github.com/mongodb/mongodb-atlas-cli/internal/log"
+	atlasauth "go.mongodb.org/atlas/auth"
+	atlas "go.mongodb.org/atlas/mongodbatlas"
+	atlasv2 "go.mongodb.org/atlas/mongodbatlasv2"
+	"go.mongodb.org/ops-manager/opsmngr"
+)
+
+const (
+	yes                       = "yes"
+	responseHeaderTimeout     = 1 * time.Minute
+	telemetryTimeout          = 1 * time.Second
+	tlsHandshakeTimeout       = 5 * time.Second
+	timeout                   = 5 * time.Second
+	keepAlive                 = 30 * time.Second
+	maxIdleConns              = 5
+	maxIdleConnsPerHost       = 4
+	idleConnTimeout           = 30 * time.Second
+	expectContinueTimeout     = 1 * time.Second
+	versionManifestStaticPath = "https://opsmanager.mongodb.com/"
+	cloudGovServiceURL        = "https://cloud.mongodbgov.com/"
+)
+
+var errUnsupportedService = errors.New("unsupported service")
+
+type Store struct {
+	service       string
+	baseURL       string
+	caCertificate string
+	skipVerify    bool
+	telemetry     bool
+	username      string
+	password      string
+	accessToken   *atlasauth.Token
+	client        interface{}
```

Comment:
if possible let's have only SDKv2 and no old client

Reply:
It would be done in follow-up PR. I intentionally did not wanted to migrate to v2 client on this as it would make review much harder.

Reply:
Done. Left 2 clients for now as we going to migrate things after merging to master. 
I did left jira to remove v1 client once migration finishes so this will be addressed eventually.